### PR TITLE
# update: use filepath.Separator instead of char "/" for cross platform

### PR DESCRIPTION
**Pull request by likakuli**
**discuss page**: https://discuss.elastic.co/t/logpathmatcher-cannot-work-correctly-because-of-the-os-specified-char/131409

### CODE REVIEW

**Comment by jsoriano** [comment]
@likakuli thanks for this PR, are you using filebeat with k8s in Windows?
I wonder if we should also change other default paths in this file for cross platform.

**Comment by jsoriano** [comment]
Ok, In any case we could have different defaults for these paths depending on the OS.
I'm fine with merging this as is by now :)

**Comment by jsoriano** [comment]
jenkins, test this, please

**Comment by likakuli** [comment]
@exekias I have updated the unit test.

**Comment by likakuli** [comment]
@jsoriano Is there any problems?

File: filebeat/processor/add_kubernetes_metadata/matchers_test.go [hunk 1]
```diff
@@ -36,20 +37,29 @@ func TestLogsPathMatcher_InvalidSource3(t *testing.T) {
 func TestLogsPathMatcher_VarLibDockerContainers(t *testing.T) {
 	cfgLogsPath := "" // use the default matcher configuration
 	source := fmt.Sprintf("/var/lib/docker/containers/%s/%s-json.log", cid, cid)
+	if runtime.GOOS == "windows" {
+		source = fmt.Sprintf("/var/lib/docker/containers/\\%s/%s-json.log", cid, cid)
```
[the remark this answers is not in the document]
Comment (likakuli) [reply filebeat/processor/add_kubernetes_metadata/matchers_test.go]: Sure these paths are not proper for windows, i changed the default logpath for windows in unit test, i think this should be OK now.

File: filebeat/processor/add_kubernetes_metadata/matchers.go [hunk 2]
```diff
@@ -39,8 +40,9 @@ func newLogsPathMatcher(cfg common.Config) (add_kubernetes_metadata.Matcher, err
 	}
 
 	logPath := config.LogsPath
-	if logPath[len(logPath)-1:] != "/" {
-		logPath = logPath + "/"
+	separator := string(filepath.Separator)
+	if logPath[len(logPath)-1:] != separator {
```
[the remark this answers is not in the document]
Comment (likakuli) [reply filebeat/processor/add_kubernetes_metadata/matchers.go]: I agress! Thanks, I had change it

File: filebeat/processor/add_kubernetes_metadata/matchers_test.go [hunk 3]
```diff
@@ -36,20 +37,32 @@ func TestLogsPathMatcher_InvalidSource3(t *testing.T) {
 func TestLogsPathMatcher_VarLibDockerContainers(t *testing.T) {
 	cfgLogsPath := "" // use the default matcher configuration
 	source := fmt.Sprintf("/var/lib/docker/containers/%s/%s-json.log", cid, cid)
+	if runtime.GOOS == "windows" {
+		cfgLogsPath = "D:\\containers\\logs"
```
Comment (jsoriano) [inline filebeat/processor/add_kubernetes_metadata/matchers_test.go]: Here the test is testing the default value, so this variable should be the empty string.

File: filebeat/processor/add_kubernetes_metadata/matchers_test.go [hunk 4]
```diff
@@ -36,20 +37,32 @@ func TestLogsPathMatcher_InvalidSource3(t *testing.T) {
...
+	}
 	expectedResult := cid
 	executeTest(t, cfgLogsPath, source, expectedResult)
 }
 
 func TestLogsPathMatcher_AnotherLogDir(t *testing.T) {
 	cfgLogsPath := "/var/log/other/"
 	source := fmt.Sprintf("/var/log/other/%s.log", cid)
+	if runtime.GOOS == "windows" {
+		cfgLogsPath = "D:\\containers\\logs"
```
Comment (jsoriano) [inline filebeat/processor/add_kubernetes_metadata/matchers_test.go]: Here the test is testing "another" log directory, so this should be different to the previous one.

File: filebeat/processor/add_kubernetes_metadata/matchers_test.go (hunk 3 above)
Comment (jsoriano) [reply filebeat/processor/add_kubernetes_metadata/matchers_test.go]: For this I guess that a different default has to be added in the mapper for windows.

**Comment by exekias** [comment]
Closing as this was already implemented long time ago

**Pull request closed without merging**